### PR TITLE
feat: Pagination variant

### PR DESCRIPTION
**Pull request by ccedrone**
Issue: # [991](https://github.com/cengage/react-magma/issues/991)

## What I did
- New pagination variant adapting pagination into a dropdown selector.

## Screenshots (if appropriate):
hooray

## Checklist 
- [x] changeset has been added
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test

- Make sure the type `PaginationType.simple` works as a dropdown with two navigation buttons.
- Ensure the doc site has the appropriate changes.

### CODE REVIEW

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        fieldId={id}
+        isInverse={isInverse}
+        onChange={handleChange}
+        theme={theme}
+        value={selectedValue}
+      >
+        {Array.from({ length: count }, (_, i) => (
+          <option
+            data-testid={`${testId}-option`}
+            key={i + count}
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:142]: Why `i + count`? This can just be `i`

File: packages/react-magma-dom/src/components/Pagination/Pagination.stories.tsx [hunk 2]
```diff
@@ -10,13 +9,35 @@ const Template: Story<PaginationProps> = args => <Pagination {...args} />;
...
+    hidePreviousButton: {
+      control: {
+        type: 'boolean',
+      },
+    },
+    hideNextButton: {
+      control: {
+        type: 'boolean',
+      },
+    },
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/Pagination.stories.tsx:40]: Let's add the simple prop here

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            onChange={handleChange}
+            value={i + 1}
+          >
+            {i + 1}
+          </option>
+        ))}
+      </StyledNativeSelect>
+      <Spacer size={8} />
+      <label aria-hidden="true">of {count} pages</label>
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:151]: We should i18n these. Also, we need to account for when `count` is 1, it should say `page`

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  }
+`;
+
+const StyledNativeSelect = styled(NativeSelect)<{
+  disabled?: boolean;
+}>`
+  select {
+    color: ${props => (props.disabled ? props.theme.colors.neutral500 : '')};
+    cursor: ${props => (props.disabled ? 'not-allowed' : '')};
+  }
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:48]: Just curious why this doesn't work by default with our NativeSelects?

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      <Pagination simple count={4} defaultPage={3} testId={testId} />
+    );
+    let nextButton = getByLabelText('Next Page');
+
+    fireEvent.click(nextButton);
+
+    expect(nextButton).toHaveStyleRule(
+      'color',
+      transparentize(0.4, magma.colors.neutral500)
+    );
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:117]: In addition to checking the color changes, can we check that the element is disabled?

File: packages/react-magma-dom/src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -118,7 +124,7 @@ function BuildButtonSize(props) {
   }
 }
 
-const NavButton = styled(IconButton)`
+export const NavButton = styled(IconButton)`
```
[the remark this answers is not in the document]
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/Pagination.tsx:127]: Indeed, it's used in 'SimplePagination'.

**Comment by silvalaura** [comment]
@orion-cengage Do we want to support both size medium and large for Simple? 
The large currently looks odd, maybe because the text is too small, maybe because it's odd for the arrows to be that large?
image

File: packages/react-magma-dom/src/components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -65,6 +66,11 @@ export interface BasePaginationProps
    * @default false
    */
   showLastButton?: boolean;
+  /**
+   * If true, shows the Pagination pages in a dropdown style.
+   * @default false
+   */
+  simple?: boolean;
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/Pagination.tsx:73]: **Question for consideration:**
In other components when we have different types/versions, we use a `type` prop, and we set a default (example: Button, LoadingIndicator, List, Input). Should we match that for consistency, or is it better to have a boolean prop?

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+  it('Should change the active page when an option is selected', () => {
+    const { getByTestId, getAllByTestId } = render(
+      <Pagination simple count={4} testId={testId} />
+    );
+    fireEvent.change(getByTestId(`${testId}-select`), { target: { value: 2 } });
+    let options = getAllByTestId(`${testId}-option`);
+    expect(options[0].selected).toBeFalsy();
+    expect(options[1].selected).toBeTruthy();
+    expect(options[2].selected).toBeFalsy();
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:140]:
```suggestion
    expect(options[0].selected).toBe(false);
    expect(options[1].selected).toBe(true);
    expect(options[2].selected).toBe(true);
```

_[source](**url**)_

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+    expect(previousButton).toHaveStyleRule(
+      'color',
+      transparentize(0.7, magma.colors.neutral100)
+    );
+    expect(nextButton).toHaveStyleRule(
+      'color',
+      transparentize(0.7, magma.colors.neutral100)
+    );
+  });
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:90]: Just a style recommendation, but I like using `describe`s to group together similar tests, or tests related to the same prop. So these could be wrapped in
```
describe('disabled', () => { 
   it ('', () => {});
   it ('', () => {});
})
```
I think it helps organize them and it's easier to see props have what coverage. I did this in ButtonGroup tests, for example.

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+
+  it('Does not violate accessibility standards', () => {
+    const { container } = render(<Pagination simple count={4} />);
+
+    return axe(container.innerHTML).then(result => {
+      return expect(result).toHaveNoViolations();
+    });
+  });
+});
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:178]: Lets add tests for edge cases:
- what if defaultPage or count is 0? -> what do we expect to happen?
- what if defaultPage or count is 1? 

What about coverage for the other Pagination props: hidePreviousButton, numberOfAdjacentPages, etc etc.

**Comment by ccedrone** [comment]
> @orion-cengage Do we want to support both size medium and large for Simple? The large currently looks odd, maybe because the text is too small, maybe because it's odd for the arrows to be that large? image

Removed per Orion, just the medium size will remain.

File: packages/react-magma-dom/src/components/Pagination/Pagination.tsx (hunk 7 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/Pagination.tsx:73]: Converted to an enum for consistency!

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx (hunk 4 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:48]: All set, removed those repetitive attributes.

File: packages/react-magma-dom/src/components/Pagination/Pagination.tsx [hunk 11]
```diff
@@ -65,6 +66,16 @@ export interface BasePaginationProps
...
   showLastButton?: boolean;
+  /**
+   * If true, shows the Pagination pages in a dropdown style.
+   * @default button
+   */
+  type?: PaginationType;
+}
+
+export enum PaginationType {
+  default = 'button',
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/Pagination.tsx:77]: I think we should call the types `simple` and `classic` (at least that's what Orion used in the designs)
```suggestion
  classic = 'classic',
```

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+      const nextButton = getByLabelText('Next Page');
+
+      expect(nextButton).toHaveStyleRule(
+        'color',
+        transparentize(0.4, magma.colors.neutral500)
+      );
+    });
+  });
+
+  describe('Active', () => {
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:149]: These look great 👏

File: website/react-magma-docs/src/pages/api/pagination.mdx [hunk 13]
```diff
@@ -28,6 +28,19 @@ export function Example() {
 }
 ```
 
+## Simple Pagination
+
+An alternative presentation of `Pagination` is availble with the `simple` prop. This converts the existing structure into a dropdown format, while still utilizing previous and next navigation buttons.
```
[the remark this answers is not in the document]
Comment (silvalaura) [reply website/react-magma-docs/src/pages/api/pagination.mdx:33]: Also, this would be a good place to point out what props can't be used with this type

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+    it('Should disable the previous button on the first page', () => {
+      const { getByLabelText } = render(
+        <Pagination type={PaginationType.simple} count={4} defaultPage={1} />
+      );
+      const previousButton = getByLabelText('Previous Page');
+
+      expect(previousButton).toHaveStyleRule(
+        'color',
+        transparentize(0.4, magma.colors.neutral500)
+      );
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:133]: Instead of checking the colors, we could expect the button `.toBeDisabled()`. The color checking should happen in the Button tests because that's part of the button component. The logic in this component just cares that the element became disabled. 

_(same feedback below for next button)_

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js (hunk 10 above)
Comment (silvalaura) [reply packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:290]: > Lets add tests for edge cases:
> 
> * what if defaultPage or count is 0? -> what do we expect to happen?
> * what if defaultPage or count is 1?
> 
> What about coverage for the other Pagination props: hidePreviousButton, numberOfAdjacentPages, etc etc.

Looks like the tests for the extra props have been addressed, but would still love to see more unit tests from the first two points, especially since you are now handling the singular/plural "pages" label change

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+  display: flex;
+  align-items: center;
+  max-width: 360px;
+  select {
+    min-width: 87px;
+  }
+  label {
+    color: ${buildLabelColor};
+    font-size: ${props => props.theme.typeScale.size02.fontSize};
+    font-weight: 500;
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:41]: I recently went through and added font-family to every component so for consistency, do you mind adding it here too?

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js (hunk 10 above)
Comment (silvalaura) [reply packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:290]: Oooh, and tests that when a page gets changed (1. from previous button 2. from next button 3. from dropdown), the `onPageChange` function gets called

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+  function paginationLabel() {
+    if (count <= 1) {
+      return `${i18n.simplePagination.ofLabel}
+        ${count}
+        ${i18n.simplePagination.pageLabel}`;
+    } else {
+      return `${i18n.simplePagination.ofLabel}
+        ${count}
+        ${i18n.simplePagination.pagesLabel}`;
+    }
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:101]: This works great and you can just leave it. But **if** you want an extra little challenge, can you think of another way to write this with less code repeat? hint: it can be done in a single line :)

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+      isInverse={isInverse}
+      ref={ref}
+    >
+      {!hidePreviousButton && (
+        <StyledPrevTooltip content={i18n.pagination.previousButtonLabel}>
+          <NavButton
+            aria-label={i18n.pagination.previousButtonLabel}
+            variant={ButtonVariant.link}
+            color={ButtonColor.secondary}
+            disabled={selectedPage < 2 ? true : disabled}
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:134]: This looks like the same logic in the `disabledPrevTooltip` variable. Can that be used here?

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+      </VisuallyHidden>
+
+      <Spacer size={14} />
+      {!hideNextButton && (
+        <StyledNextTooltip content={i18n.pagination.nextButtonLabel}>
+          <NavButton
+            aria-label={i18n.pagination.nextButtonLabel}
+            variant={ButtonVariant.link}
+            color={ButtonColor.secondary}
+            disabled={selectedPage >= count ? true : disabled}
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:179]: Same note as above but `disabledNextTooltip`

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+            disabled={selectedPage < 2 ? true : disabled}
+            icon={<ArrowBackIcon />}
+            isInverse={isInverse}
+            theme={theme}
+            onClick={handlePrev}
+            shape={ButtonShape.fill}
+          />
+        </StyledPrevTooltip>
+      )}
+      <Spacer size={14} />
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:143]: If `hidePreviousButton` is true, should we have this Spacer? or should we wrap this as part of that logic?

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+        data-testid={`${testId}-select`}
+        disabled={disabled}
+        fieldId={id}
+        isInverse={isInverse}
+        onChange={handleChange}
+        value={selectedPage}
+      >
+        {Array.from({ length: count }, (_, i) => (
+          <option
+            data-testid={`${testId}-option`}
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:156]: This needs to be unique:
```suggestion
            data-testid={`${testId}-option-${i}`}
```

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:1]: I'm not sure how common of a use case this is, but I wonder if we need to account for the possibility of the `count` changing. I am doing it on Storybook and that's how I ran into this, but I'm thinking that perhaps API data will load and may need to update the count of the component. If so, I think we need to reset `selectedPage` anytime the value of count changes 🤔  we can talk through this one, I'm just seeing the `selectedPage` get out of sync in some edge cases.

File: packages/react-magma-dom/src/components/Pagination/Pagination.tsx (hunk 11 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/Pagination.tsx:77]: Should this be renamed in the files as well?

File: website/react-magma-docs/src/pages/api/pagination.mdx [hunk 21]
```diff
@@ -28,6 +28,19 @@ export function Example() {
...
+
+```typescript
+import React from 'react';
+import { Pagination } from 'react-magma-dom';
+
+export function Example() {
+  return <Pagination count={4} type="simple" />;
```
[the remark this answers is not in the document]
Comment (ccedrone) [reply website/react-magma-docs/src/pages/api/pagination.mdx:40]: Don't we just want `type="simple"` instead?

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js (hunk 12 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/SimplePagination.test.js:149]: 🕺

File: website/react-magma-docs/src/pages/api/pagination.mdx (hunk 21 above)
[the remark this answers is not in the document]
Comment (silvalaura) [reply website/react-magma-docs/src/pages/api/pagination.mdx:40]: It "works", but really the `type` prop expects an Enum, so `PaginationType.simple` is preferred. Here's a similar example where you can see CodeSandbox complain: 
image

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx (hunk 19 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:143]: That's a good point, moved it inside the conditional!

File: packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx (hunk 16 above)
Comment (ccedrone) [reply packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx:101]: Accepted!

**Review by silvalaura — COMMENTED** [review]
A couple more notes:
- MUI supports `count={0}`, so I do think it's important we do that (0 vs 1 vs 10 below) because our classic type does.
image

- In the [Pagination Ranges](https://mui.com/material-ui/react-pagination/#pagination-ranges) example, it shows that behavior I was mentioning about "resetting" the component when the count changes: the defaultPage is 2, I clicked through to page 6, then changed the count and that reset the selectedPage to 2.
chrome-capture-2023-2-21 (1)

**Review by silvalaura — COMMENTED** [review]
Looks great! 

One final note, I noticed this in the docs: 
image

This is because `PaginationVariant` is not exported in `react-magma/packages/react-magma-dom/src/index.ts`.


_Bonus but not required:_
I really like this example because it "proves" how it works/how the component could be hooked up in a real application. Maybe it's worth adding that functionality to the simple example?
image

**Comment by silvalaura** [comment]
This PR is more than a draft, I think it's ready to be set for review ;) Also, don't forget to add the description, screenshots and testing notes.

**Review by katieweinstein — APPROVED** [review]

**Review by silvalaura — APPROVED** [review]

**Review by orion-cengage — APPROVED** [review]
